subtractFlats.py

```diff
@@ -115,80 +104,93 @@ def main():
     suffix       = '_f2_R22_S11_E000.fits.gz'
 
     # Process Files
-    magnitude = ['18', '15', '14', '13', '12', '10']
+    magnitude = [18, 15, 14, 13, 12, 10]
     extraId   = ['0', '1', '2', '3', '4']
 
```

Comment:
* "magnitude" is a well-ordered variable with a clear associated physical meaning.  Thus declaring it to be a number seemed to satisfy those goals. However, I kept it as an integer rather that a float, so that it could be used as a hash key (dict in Python) for lookups with no concern about float equality/representation.

* "extraId" seemed more like a list of labels

One could imagine.

extraId   = ['0', '1', '2', '3', '4', 'test', 'panda']

and everything should still work.

I, of course, could have reasonably assumed that "extraId" was really an index into a more descriptive array.  And, if I had access to that information, I might have included that descriptive array in the code just for reference.  E.g.,

tests = ["test_foo", "test_bar", "test_bar_foo", "test_panda", "test_pie"]
for (eid, t) in enumerate(tests):

or if it were useful to keep around

extraId = range(len(tests))

However, even then, there wouldn't necessarily be a clear ordering, and the variable is likely not continuous, despite the implied ordering of an enumerate or range generation.

But fundamentally, I just didn't know enough about the plans for extraId to know whether it was a list of labels or meant to be an key index into something else.  In particular, because it didn't appear to have an ordering or continuity, keeping it as a string seemed most appropriate.

 - Michael

> On Mar 26, 2015, at 20:41 , Chris Walter <notifications@github.com> wrote:
> 
> In subtractFlats.py:
> 
> > @@ -115,80 +104,93 @@ def main():
> >      suffix       = '_f2_R22_S11_E000.fits.gz'
> >  
> >      # Process Files
> > -    magnitude = ['18', '15', '14', '13', '12', '10']
> > +    magnitude = [18, 15, 14, 13, 12, 10]
> >      extraId   = ['0', '1', '2', '3', '4']
> >  
> 
> I've learned a lot of interesting python tidbits from your changes!
> 
> Why did you change the magnitudes from strings to integers but not the extraId here?
> 
> —
> Reply to this email directly or view it on GitHub.
>